Update deprecations for 0.33 (#4720)
PennyLaneAI/pennylane: doc/development/deprecations.rst

Author: The diff is a bit hard to see, I recommend viewing in the docs (once built).

* ``qml.ExpvalCost`` has been deprecated, and usage will now raise a warning.
  
  - Deprecated in v0.24

>   - Will be removed in v0.35

Reviewer: If this is the case we should probably update `ShotAdaptiveOptimizer` for 0.34. A big problem with the optimizer was that when using `ExpvalCost`, it would split each Hamiltonian term into its own QNode and allocate shots. Now that we have shots on the tape, we can update this behaviour.

Author: Yep agreed, `ShotAdaptiveOptimizer` is in the 0.34 plan.

  `discussion forum <https://discuss.pennylane.ai/>`_.

  - Deprecated in v0.31
  - Changed in v0.33

> * The behaviour of ``Operator.__eq__`` and ``Operator.__hash__`` will be updated soon. Their documentation

Author: ```suggestion
* The behaviour of ``Operator.__eq__`` and ``Operator.__hash__`` has been updated. Their documentation
```

>   has been updated to reflect the incoming changes.

Reviewer: The old behaviour has been removed. 0.33 has `==` behaving the same as `qml.equal`.

>   The upcoming changes to operator equality will allow users to use operator equality the same way as
>   with ``qml.equal``. With the changes to hashing, unique operators that are equal will have the same
>   hash. These changes will allow behaviour such as the following:

Reviewer: ```suggestion
  The changes to operator equality allow users to use operator equality the same way as
  with ``qml.equal``. With the changes to hashing, unique operators that are equal now have the same
  hash. These changes now allow behaviour such as the following:
```
